docs/tutorials/adding-a-module-to-your-runtime.md

```diff
@@ -0,0 +1,306 @@
+
+In the case of the Contracts module, we actually want a hook when an account runs out of free balance. Because the Contract module instantiates contracts as "Accounts", it also needs to know when an account is destroyed so that it can clean up any storage that contract was using. You can find that logic in the Contract module source code:
+
+**Contract module `lib.rs`**
+```rust
+impl<T: Trait> OnFreeBalanceZero<T::AccountId> for Module<T> {
+	fn on_free_balance_zero(who: &T::AccountId) {
+		<CodeHashOf<T>>::remove(who);
+		<AccountInfoOf<T>>::get(who).map(|info| child::kill_storage(&info.trie_id));
+	}
+}
+```
+
+To enable this, we simply need to add `Contract` to the `OnFreeBalanceZero` hook provided by the Balances module:
+
+**`contract-chain/runtime/src/lib.rs`**
+```rust
+impl balances::Trait for Runtime {
+    /// What to do if an account's free balance gets zeroed.
+    type OnFreeBalanceZero = (Contract);
+    ...
+}
+```
+
+Now, when the Balances module detects that the free balance of an account has reached zero, it calls the `on_free_balance_zero` function of the contract module.
+
+## Genesis Configuration
+
+The last thing we need to do in order to get your node up and running is to establish a genesis configuration for the Contract module. This is controlled in `contract-chain/src/chain_spec.rs`. We need to modify this file to include the `ContractConfig` type:
+
+```rust
+use contract_chain_runtime::ContractConfig;
+```
+
+Then we need to add values to the configuration. As of Substrate v1.0, it looks like:
+
+```rust
+let mut contract_config = ContractConfig {
+    transaction_base_fee: 1,
+    transaction_byte_fee: 0,
+    transfer_fee: 0,
+    creation_fee: 0,
+    contract_fee: 21,
+    call_base_fee: 135,
+    create_base_fee: 175,
+    gas_price: 1,
+    max_depth: 1024,
+    block_gas_limit: 10_000_000,
+    current_schedule: Default::default(),
+};
+// IMPORTANT: this should only be enabled on development chains!
+contract_config.current_schedule.enable_println = true;
+```
+
+Note that you can tweak these numbers to your needs, but these are the values set at the genesis configuration of the main Substrate node. You should place that code block into the `testnet_genesis` function, and then in the `GenesisConfig` object, add:
+
+```rust
+GenesisConfig {
+    ...
+    contract: Some(contract_config),
+}
+```
+
+Finally, you are ready to compile your contract capable node with:
+
+```rust
+./scripts/build.sh
+cargo build --release
+```
+
+## Next Steps
+
+We wont actually go into the details of testing the Contract module functionality added to this chain, but if you want to try it out, you can follow our tutorial on [Deploying Your First Smart Contract] (LINK TO BE ADDED).
```

Comment:
I'm 100% comfortable publishing this before the deploying tutorial is ready. Maybe just say coming soon.

Reply:
I just put the future link there. Should make it in when we merge the other PR